### DISCUSSION TITLE
Using Dynamic Graphs and Op Priority to run each batch of data from end-to-end before moving to the next

### DISCUSSION THREAD

**Discussion by vandenn** [dagster-io/dagster#16264] [Show and tell]
# Repository Link

I recently got some time to create a reference repository of how I use Dynamic Graphs and the `dagster/priority` tag in tandem to improve my job's performance and save memory:
https://github.com/vandenn/dagster-prio-dynamic-map

# Overview

When you use `DynamicOut` , you can chunk data into batches yielded by the dynamic op, but each subsequent op that `map`s to these batches executes on all of them first before moving to the next op:

![standard_dynamic](https://user-images.githubusercontent.com/6585214/265042166-34116c7e-373f-43ba-8044-783b0be47729.jpg)

However, this can be a problem if you have memory constraints in the machine you've deployed Dagster to.

By combining it with the `dagster/priority` tag, you can run the "mini-job" end-to-end for a group of batches of data (number of batches depends on your machine's power or your specified `max_concurrent` tag) before moving to the next group. This will allow you to clean/dispose already processed data as the job goes through each group of batches.

![prio_dynamic](https://user-images.githubusercontent.com/6585214/265045021-f05ae1b7-85d6-4651-8844-bc0b09830046.jpg)

# Final Note

Let me know what you think! I'd appreciate any feedback or comments.

**Comment by PedramNavid** [comment]
This is really nice, thanks for sharing this!

**Comment by chrishiste** [comment]
@vandenn You are a genius, thanks for this :)
